### PR TITLE
ZPA Data Connector

### PR DESCRIPTION
**Pull request by vu-socprime**
## Proposed Changes

  - add Zscaler Private Access data connector

### CODE REVIEW

File: DataConnectors/ZPA/Connector_LogAnalytics_agent_Zscaler_ZPA.json [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+                "requiredPermissions": {
+                    "action": true
+                }
+            }
+        ]
+    },
+    "instructionSteps": [
+        { 
+            "title": "", 
+            "description": ">**NOTE:** This data connector depends on a parser based on a Kusto Function to work as expected. [Follow these steps](https://aka.ms/sentinel-Ubiquiti-parser) to create the Kusto Functions alias, **ZPAEvent**",
```
Comment (v-admahe) [inline DataConnectors/ZPA/Connector_LogAnalytics_agent_Zscaler_ZPA.json:66]: Parser link should be "https://aka.ms/sentinel-ZPAEvent-parser".Please update the parser link.

File: Parsers/ZPA/ZPAEvent.txt [hunk 2]
```diff
@@ -0,0 +1,292 @@
...
+            HttpRequestBodyBytes=column_ifexists('RequestSize', ''),
+            HttpResponseBodyBytes=column_ifexists('ResponseSize', ''),
+            SrcPortNumber=column_ifexists('ClientPublicPort', ''),
+            EventResultDetails=column_ifexists('ConnectionReason', ''),
+            CorsToken=column_ifexists('CorsToken', ''),
+            Origin=column_ifexists('Origin', '')
+       | project
+            LogTimestamp,
+            Customer,
+            SessionID,
```
Comment (v-admahe) [inline Parsers/ZPA/ZPAEvent.txt:156]: Does this should be "NetworkSessionId". Please refer the link for naming convection https://docs.microsoft.com/en-us/azure/sentinel/normalization-schema

File: Parsers/ZPA/ZPAEvent.txt [hunk 3]
```diff
@@ -0,0 +1,292 @@
...
+            SrcGeoLatitude,
+            SrcGeoLongitude,
+            SrcGeoCountry,
+            ClientZEN,
+            NetworkRuleName,
+            Connector,
+            ConnectorZEN,
+            ConnectorIP,
+            ConnectorPort,
+            Host,
```
Comment (v-admahe) [inline Parsers/ZPA/ZPAEvent.txt:175]: Does this should be "SrcDvcHostname"? Please refer the https://docs.microsoft.com/en-us/azure/sentinel/normalization-schema

File: Parsers/ZPA/ZPAEvent.txt [hunk 4]
```diff
@@ -0,0 +1,292 @@
...
+            TrustedNetworks,
+            TrustedNetworksNames,
+            SAMLAttributes,
+            PosturesHit,
+            PosturesMiss,
+            ZENLatitude,
+            ZENLongitude,
+            ZENCountryCode,
+            SessionType,
+            DvcOS,
```
Comment (v-admahe) [inline Parsers/ZPA/ZPAEvent.txt:231]: Does this should be "SrcDvcOs"? Please refer the https://docs.microsoft.com/en-us/azure/sentinel/normalization-schema

File: Parsers/ZPA/ZPAEvent.txt [hunk 5]
```diff
@@ -0,0 +1,292 @@
...
+            CertificateCN,
+            PrivateIP,
+            DvcIpAddr,
+            Latitude,
+            Longitude,
+            CountryCode,
+            TimestampAuthentication,
+            TimestampUnAuthentication,
+            DvcHostname,
+            DvcOs,
```
Comment (v-admahe) [inline Parsers/ZPA/ZPAEvent.txt:220]: Does this should be "SrcDvcOs"? Please refer the https://docs.microsoft.com/en-us/azure/sentinel/normalization-schema

**Review by v-admahe — CHANGES_REQUESTED** [review]

File: DataConnectors/ZPA/Connector_LogAnalytics_agent_Zscaler_ZPA.json (hunk 1 above)
Comment (vu-socprime) [reply DataConnectors/ZPA/Connector_LogAnalytics_agent_Zscaler_ZPA.json]: fixed

File: DataConnectors/ZPA/Connector_LogAnalytics_agent_Zscaler_ZPA.json [hunk 6]
```diff
@@ -0,0 +1,166 @@
+{
+    "id": "Zscaler",
```
Comment (v-admahe) [inline DataConnectors/ZPA/Connector_LogAnalytics_agent_Zscaler_ZPA.json:2]: Could you please change the connector id , As we already has the connector deployed with same ID.

Thanks,
Aditi Maheshwari

**Review by v-admahe — CHANGES_REQUESTED** [review]

File: Parsers/ZPA/ZPAEvent.txt (hunk 2 above)
Comment (vu-socprime) [reply Parsers/ZPA/ZPAEvent.txt]: updated

File: Parsers/ZPA/ZPAEvent.txt (hunk 3 above)
Comment (vu-socprime) [reply Parsers/ZPA/ZPAEvent.txt]: updated

File: Parsers/ZPA/ZPAEvent.txt (hunk 4 above)
Comment (vu-socprime) [reply Parsers/ZPA/ZPAEvent.txt]: updated

File: Parsers/ZPA/ZPAEvent.txt (hunk 5 above)
Comment (vu-socprime) [reply Parsers/ZPA/ZPAEvent.txt]: updated

File: DataConnectors/ZPA/Connector_LogAnalytics_agent_Zscaler_ZPA.json (hunk 6 above)
Comment (vu-socprime) [reply DataConnectors/ZPA/Connector_LogAnalytics_agent_Zscaler_ZPA.json:2]: updated

File: DataConnectors/ZPA/Connector_LogAnalytics_agent_Zscaler_ZPA.json [hunk 7]
```diff
@@ -0,0 +1,166 @@
...
+        {
+            "metricName": "Total data received",
+            "legend": "ZPA_CL",
+            "baseQuery": "ZPA_CL"
+        }
+    ],
+    "sampleQueries": [
+        {
+            "description" : "All logs",
+            "query": "\nZPAEvent\n| sort by TimeGenerated"
```
Comment (v-jayakal) [inline DataConnectors/ZPA/Connector_LogAnalytics_agent_Zscaler_ZPA.json:18]: Invalid query, Can you please correct this? TimeGenerated is available in the Parser.

File: DataConnectors/ZPA/Connector_LogAnalytics_agent_Zscaler_ZPA.json [hunk 8]
```diff
@@ -0,0 +1,166 @@
+{
+    "id": "ZPA",
```
Comment (v-jayakal) [inline DataConnectors/ZPA/Connector_LogAnalytics_agent_Zscaler_ZPA.json:2]: Can you please expand the id? "zscalarprivateaccess" is a suggestion.

**Review by v-jayakal — CHANGES_REQUESTED** [review]

File: DataConnectors/ZPA/Connector_LogAnalytics_agent_Zscaler_ZPA.json (hunk 7 above)
Comment (vu-socprime) [reply DataConnectors/ZPA/Connector_LogAnalytics_agent_Zscaler_ZPA.json:18]: added TimeGenerated to parser

File: DataConnectors/ZPA/Connector_LogAnalytics_agent_Zscaler_ZPA.json (hunk 8 above)
Comment (vu-socprime) [reply DataConnectors/ZPA/Connector_LogAnalytics_agent_Zscaler_ZPA.json]: updated

**Review by v-jayakal — APPROVED** [review]

File: DataConnectors/ZPA/Connector_LogAnalytics_agent_Zscaler_ZPA.json [hunk 9]
```diff
@@ -0,0 +1,166 @@
...
+    "instructionSteps": [
+        { 
+            "title": "", 
+            "description": ">**NOTE:** This data connector depends on a parser based on a Kusto Function to work as expected. [Follow these steps](https://aka.ms/sentinel-ZscalarPrivateAccess-parser) to create the Kusto Functions alias, **ZPAEvent**",
+            "instructions": [ 
+            ]    
+        },
+        {
+            "title": "",
+            "description": ">**NOTE:** This data connector has been developed using Zscaler Private Access version: ____",
```
Comment (v-rucdu) [inline DataConnectors/ZPA/Connector_LogAnalytics_agent_Zscaler_ZPA.json:72]: @socprime : Can you please provide the version details for this connector?

**Pull request merged**